genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/ShutdownAction.java

```diff
@@ -45,7 +58,25 @@
     @Override
     protected Events executeStateAction(final ExecutionContext executionContext) {
         log.info("Shutting down...");
-        //TODO implement this action
+
+        try {
+            log.info("Attempting to archive job folder...");
```

Comment:
Would recommend logging this information after you know the location and including the location in the message something like `Attempting to backup job folder to {location}...`

Reply:
👍